examples/07_join_aggregation.py

```diff
@@ -0,0 +1,438 @@
+)
+
+
+# In[7]:
+
+
+user_count_ratings = (
+    ratings.groupby("userId").agg(n_ratings=("rating", "count")).reset_index()
+)
+
+print(
+    "min:",
+    user_count_ratings["n_ratings"].min(),
+    "max:",
+    user_count_ratings["n_ratings"].max(),
+)
+
+ax = sns.histplot(user_count_ratings["n_ratings"])
+ax.set(
+    xlabel="Number of ratings given by users",
+    ylabel="Total number of users",
+    title="Number of ratings given by user",
+)
+
+
+# In[8]:
+
+
+rating_count = ratings["rating"].value_counts().sort_index()
+
+make_barplot(
+    x=rating_count.index,
+    y=rating_count.values,
+    title="Distribution of ratings given to movies",
+)
+
+
+# In[9]:
+
+
+## Split train test by date
+
+min_date = ratings["timestamp"].min()
+max_date = ratings["timestamp"].max()
+duration = max_date - min_date
+train_threshold = max_date - duration / 3
+
+train = ratings.loc[ratings["timestamp"] < train_threshold]
+test = ratings.loc[ratings["timestamp"] >= train_threshold]
+train.shape, test.shape
+
+
+# ## Feature Engineering
+
+# ***Ohe Hot Encoding day names***
+
+# In[10]:
+
+
+from sklearn.pipeline import make_pipeline, FunctionTransformer
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import OneHotEncoder
+
+col_trans = make_column_transformer(
+    ("passthrough", ["userId", "movieId", "rating"]),
+    (OneHotEncoder(sparse_output=False), ["day_name"]),
+    remainder="drop",
+    verbose_feature_names_out=False,
+)
+
+train = col_trans.fit_transform(train)
+train = pd.DataFrame(train, columns=col_trans.get_feature_names_out())
+train.head()
+
+
+# ***Join aggregator on `movieId` and `userId`***
+
+# In[11]:
+
+
+from skrub import JoinAggregator
+
+cols_day_name = list(set(train.columns) - set(["userId", "movieId"]))
+
+join_agg_rating = JoinAggregator(
+    tables=[
+        (train, "userId", cols_day_name),
+        (train, "movieId", cols_day_name),
+    ],
+    main_key=["userId", "movieId"],
+    suffixes=["_user", "_movie"],
+    agg_ops=["sum", "mean"],
+)
+train = join_agg_rating.fit_transform(train[["userId", "movieId", "rating"]])
+
+cols_to_drop = [
+    col
+    for col in train.columns
+    if ("mean" in col and "day_name" in col) or ("rating_mean" in col)
+]
+train = train.drop(cols_to_drop, axis=1)
+train
+
+
+# ### Encoding movies
+
+# In[12]:
+
+
+movies = fetch_movielens(dataset_id="movies")
+movies = movies.X
+movies
+
+
+# In[13]:
+
+
+movies["genres"] = movies["genres"].str.replace("|", " ")
+all_genres = movies["genres"].str.split().explode().unique()[:-3]
+all_genres
+
+
+# In[14]:
+
+
+from sklearn.feature_extraction.text import CountVectorizer
+from skrub import MinHashEncoder
+
+vectorizer = CountVectorizer(
+    vocabulary=all_genres,
+    ngram_range=(1, 1),
+    lowercase=False,
+)
+
+n_components = 10
+min_hash_encoder = MinHashEncoder(n_components=n_components)
+
+movie_transformer = make_column_transformer(
```

Comment:
Once our TableVectorizer works well, we should be able to replace all this by the TableVectorizer (this should be a goal for skrub

Reply:
And even before, you can test it directly using 

```python
TableVectorizer(high_card_cat_transformer= min_hash_encoder = MinHashEncoder(n_components=n_components))
```
From previous benchmarks, I don't think the `CountVectorizer` in itself does a good job here.